### PR TITLE
Model Views

### PR DESCRIPTION
**Pull request by David-Lanzz**
# Hello Reviewer, I have completed my task for today and below is a summary of my work
- No linter errors
- Correct use of git flow
- Creation of .erb files to display views
- Task completed

### CODE REVIEW

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @David-Lanzz 🙋‍♀️,  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
 
You have submitted a project, that has falling linter errors. Kindly fix the Rubocop issues and re-submit for another review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            <%= image_tag (user.photo)%>
+        </div>
+        <div class='user-info'>
+             <h2><%= user.name %></h2>
+             <p>Number of Posts: <%= user.posts_counter %></p>
+        </div>
+      </a>
+    <%end%>
+    </ul>
+    </main>
```
Comment (omarsalem7) [inline app/views/users/index.html.erb]: - [ ] Kindly consider the index view page should match the given [wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_root_users.png?token=GHSAT0AAAAAACBXQZYXMUG66IFCTDHXQZ2UZEVUS3Q)

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        </div>
+        <div>
+  <p><%=post.comments_counter %></p>
+  <p><%=post.likes_counter %></p>
+        </div>
+<% end %>
+        </ul>
+        <%end%>
+    </li>
+    </ul>
```
Comment (omarsalem7) [inline app/views/users/show.html.erb]: - [ ] Kindly consider the show page to match with the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_page.png) guidelines
- [ ] Kindly consider adding the `See all posts` link to navigate to user posts

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                 <div>
+                  <p>Comments: <%= post.comments_counter %></p>
+                  <p>Likes: <%= post.likes_counter %></p>
+                 </div>
+                 <div>
+    <% post.comments.each do |comment| %>
+        <% if comment.author_id == user.id %>
+        <%= user.name %>:<%= comment.text %> <br>
+        <% end %>
+      <% end %>
```
Comment (omarsalem7) [inline app/views/posts/index.html.erb]: - [ ] Kindly consider rendering 5 most recent comments by using `recent_comments` method in the post model and also consider following design guidelines for the [index](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_all_posts.png?token=GHSAT0AAAAAACBXQZYXKV5PPM2GFX2GYITAZEVU2YA), [show](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_single_post.png?token=GHSAT0AAAAAACBXQZYWZGBQD6GWRBKOV3EGZEVU4TA) post

File: config/routes.rb [hunk 4]
```diff
@@ -3,4 +3,7 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
```
Comment (omarsalem7) [inline config/routes.rb:5]: - [ ] Kindly consider defining the root path to `users#index` , so when anyone opens your application renders the user page not the default page for ruby on rails

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @David-Lanzz 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights

- Controller actions added properly ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 4 above)
Comment (KDlamini) [reply config/routes.rb:5]: Well resolved ✔️

File: app/views/users/index.html.erb (hunk 1 above)
Comment (KDlamini) [reply app/views/users/index.html.erb]: Well resolved ✔️ 

image

File: app/views/users/show.html.erb (hunk 2 above)
Comment (KDlamini) [reply app/views/users/show.html.erb]: Well resolved ✔️

image

File: app/views/posts/index.html.erb (hunk 3 above)
Comment (KDlamini) [reply app/views/posts/index.html.erb]: Well resolved ✔️

**Review by KDlamini — APPROVED** [review]
### Hi @David-Lanzz 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes for the Blog app views. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## Highlights 🌞

✔️  Good PR title and description
✔️  Users controller `#index` action renders all users to their corresponding Views index page
✔️  The `#recent_post` method renders the user's 3 most recent posts.
✔️  `See all Post` link opens the user's post index as displayed on the wireframe
✔️  Post show page displays post with all comments

# Status: Approved ✔️ 

LeonardoInGatsby

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**